docker/manager/_internal/_gen.py

```diff
@@ -0,0 +1,309 @@
+                "arch_context": arch_context,
+            }
+
+            cuda_target_arch = {
+                "amd64": "x86_64",
+                "arm64v8": "sbsa",
+                "ppc64le": "ppc64le",
+            }
+
+            for paths in release_tags.values():
+                for tpl_file in paths["input_paths"]:
+                    cached_key = f"{build_info.shared_context.distro_name}{ctx._templates_dir.getsyspath(tpl_file)}"
+                    if cached_key in cached:
+                        continue
+                    else:
+                        cached.append(cached_key)
+                        to_render = partial(
+                            render_template,
+                            input_name=tpl_file,
+                            inp_fs=ctx._templates_dir,
+                            output_path=paths["output_path"],
+                            out_fs=ctx._generated_dir,
+                            build_tag=paths["build_tag"],
+                            cuda=cattr.unstructure(cuda),
+                            metadata=metadata,
+                            distros=build_info.shared_context.distro_name,
+                            xx_image="tonistiigi/xx",
+                            xx_version="1.1.0",
+                            context_path=paths["output_path"],
+                        )
+
+                        try:
+                            if "rhel" in shared_context.templates_dir:
+                                cuda_supported_arch = [
+                                    k for k in arch_context if hasattr(cuda_context, k)
+                                ]
+                                for sa in cuda_supported_arch:
+                                    to_render(arch=sa, cuda_url=cuda_target_arch[sa])
+                            else:
+                                to_render()
+                        except Exception as e:  # pylint: disable=broad-except
+                            send_log(
+                                f"Error while generating Dockerfiles:\n{e}",
+                                _manager_level=logging.ERROR,
+                            )
+                            raise
+
+
+def get_python_version_from_tag(tag: str) -> t.List[int]:
+    return [int(u) for u in tag.split(":")[-1].split("-")[1].strip("python").split(".")]
+
+
+def gen_readmes(ctx: Environment) -> None:
+    tag_ref = defaultdict(list)
+    arch = {}
+
+    for distro, distro_info in ctx.release_ctx.items():
+        results = [
+            (k, v["git_tree_path"])
+            for rltx in distro_info
+            for k, v in rltx.release_tags.items()
+            if "base" not in k
+        ]
+
+        (k_sort := [t[0] for t in results]).sort(
+            key=lambda tag: get_python_version_from_tag(tag)
+        )
+        order = {k: v for v, k in enumerate(k_sort)}
+
+        tag_ref[distro] = sorted(results, key=lambda k: order[k[0]])
```

Comment:
Why sort twice here?

```suggestion
        def result_sort_key(result):
            return get_python_version_from_tag(result[0])

        tag_ref[distro] = sorted(results, key=result_sort_key)
```